model/import/SampleTemplateDownload.php

```diff
@@ -0,0 +1,115 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiItem\model\import;
+
+use oat\taoQtiItem\model\import\Repository\CsvTemplateRepository;
+use oat\taoQtiItem\model\import\Repository\MetadataRepository;
+use oat\taoQtiItem\model\import\Repository\TemplateRepositoryInterface;
+use oat\taoQtiItem\model\import\Parser\TemplateHeaderParser;
+use oat\taoQtiItem\model\import\Factory\CsvTemplateSampleLineFactory;
+use oat\oatbox\service\ConfigurableService;
+use Zend\ServiceManager\ServiceLocatorAwareInterface;
+use oat\generis\model\OntologyAwareTrait;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use function GuzzleHttp\Psr7\stream_for;
+use common_exception_Error;
+use common_exception_MissingParameter;
+
+class SampleTemplateDownload extends ConfigurableService implements ServiceLocatorAwareInterface
+{
+    use OntologyAwareTrait;
+
+    /**
+     * @throws common_exception_Error
+     */
+    public function download(ServerRequestInterface $request, ResponseInterface $response)
+    {
+        $queryParams = $request->getQueryParams();
+        if (!$queryParams['uri']) {
+            throw new common_exception_MissingParameter('uri', __METHOD__);
+        }
+        $uri = $queryParams['uri'];
+        $metaDataArray = $this->getMetadataRepository()->findMetadataByClassUri($uri);
+    
+        $template = $this->getTemplateRepository()->findById(CsvTemplateRepository::DEFAULT);
+        $headers = $this->getTemplateHeaderParser()->parse($template, $metaDataArray);
+
+        $templateSampleLines = $this->getTemplateSampleLines()->createMultiple($template);
+
+        $className = $this->getClassName($uri);
+        $filename = $this->getFileName($className);
+
+        $csvContent = $this->getCsvContent($headers, $templateSampleLines);
+        return $response
+            ->withHeader('Content-Type', 'text/csv')
+            ->withHeader("Content-Disposition", "attachment; filename=" . $filename)
+            ->withBody(stream_for($csvContent));
+    }
+
+    private function getCsvContent($headers, $templateSampleLines): string
+    {        
+        $sampleLines = implode(",",$headers).PHP_EOL;
+        foreach ($templateSampleLines as $row) {
+            $line = str_replace('"', '""',implode(","."\t",$row));
+            $sampleLines .= '"'.str_replace(","."\t", '","', $line).'"'.PHP_EOL;
+        }
+        return $sampleLines;
```

Comment:
Please, simplify this code :) 

Also pay attention to PSR-12 rules. It is missing a lot of spaces separator and break lines in your code

```suggestion
        $sampleLines = implode(",", $headers) . PHP_EOL;
        
        foreach ($templateSampleLines as $row) {
            $sampleLines .= '"' . implode('", "', $row) . '"' . PHP_EOL;
        }
        
        return $sampleLines;
```

Reply:
Done formatting